pages/studentAssessmentInstanceExam/studentAssessmentInstanceExam.sql

```diff
@@ -28,3 +28,42 @@ WHERE
 WINDOW
     w AS (ORDER BY qo.row_order)
 ORDER BY qo.row_order;
+
+-- BLOCK get_group_info
+SELECT
+    gu.group_id, g.name, g.join_code, u.uid, gc.student_authz_join, gc.student_authz_create, gc.student_authz_leave
+FROM
+    assessment_instances ai
+    JOIN group_configs AS gc ON ai.assessment_id = gc.assessment_id
+    JOIN groups AS g ON g.group_config_id = gc.id
+    JOIN group_users AS gu ON gu.group_id = g.id
+    JOIN group_users AS gu2 ON gu2.group_id = gu.group_id
+    JOIN users AS u ON u.user_id = gu2.user_id
+WHERE
+    ai.id = $assessment_instance_id
+    AND gu.user_id = $user_id
+    AND g.deleted_at IS NULL
+    AND gc.deleted_at IS NULL;
+
+-- BLOCK leave_group
+WITH log AS (
+    DELETE FROM
+        group_users
+    WHERE
+        user_id = $user_id
+        AND group_id IN (SELECT
+                            g.id
+                        FROM
+                            assessment_instances ai
+                            JOIN group_configs AS gc ON gc.assessment_id = ai.assessment_id
+                            JOIN groups AS g ON g.group_config_id = gc.id
+                        WHERE
+                            ai.id = $assessment_instance_id
+                            AND g.deleted_at IS NULL
+                            AND gc.deleted_at IS NULL)
+    RETURNING group_id
+)
+INSERT INTO group_logs
+    (authn_user_id, user_id, group_id, action)
+SELECT $authn_user_id, $user_id, group_id, 'leave'
+FROM log;
```

Comment:
As a rule, file should have a trailing newline.

```suggestion
FROM log;

```

If you see the red symbol on the last line of a GitHub diff, that means one is missing!

<img width="241" alt="Screen Shot 2022-04-12 at 15 35 20" src="https://user-images.githubusercontent.com/1476544/163065815-fad34c08-43c5-404d-a2d4-317e812aa91b.png">

Reply:
Ahh good to know about the red symbol! Fixed in [6eb93eb](https://github.com/PrairieLearn/PrairieLearn/pull/5290/commits/6eb93eb9cffa0ed27b6aef49363941d7d3492838)